modules/alloy/src/main/ts/ephox/alloy/positioning/view/Bounder.ts

```diff
@@ -165,34 +197,27 @@ const attempt = (candidate: SpotInfo, width: number, height: number, bounds: Box
 const attempts = (candidates: AnchorLayout[], anchorBox: AnchorBox, elementBox: AnchorElement, bubbles: Bubble, bounds: Boxes.Bounds): RepositionDecision => {
   const panelWidth = elementBox.width;
   const panelHeight = elementBox.height;
-  const attemptBestFit = (layout: AnchorLayout, reposition: RepositionDecision, deltaW: number, deltaH: number) => {
+  const attemptBestFit = (layout: AnchorLayout, reposition: RepositionDecision, visibleW: number, visibleH: number, isVisible: boolean) => {
     const next: SpotInfo = layout(anchorBox, elementBox, bubbles);
     const attemptLayout = attempt(next, panelWidth, panelHeight, bounds);
 
-    // unwrapping fit only to rewrap seems... silly
-    return attemptLayout.fold(adt.fit, (newReposition, newDeltaW, newDeltaH) => {
-      // console.log(`label: ${next.label()}, newDeltaW: ${newDeltaW}, deltaW: ${deltaW}, newDeltaH: ${newDeltaH}, deltaH: ${deltaH}`);
-      const improved = newDeltaH > deltaH || newDeltaW > deltaW;
+    return attemptLayout.fold(Fun.constant(attemptLayout), (newReposition, newVisibleW, newVisibleH, newIsVisible) => {
+      // console.log(`label: ${next.label}, newVisibleW: ${newVisibleW}, visibleW: ${visibleW}, newVisibleH: ${newVisibleH}, visibleH: ${visibleH}, newIsVisible: ${newIsVisible}, isVisible: ${isVisible}`);
+      const improved = isVisible === newIsVisible ? (newVisibleH > visibleH || newVisibleW > visibleW) : (!isVisible && newIsVisible);
```

Comment:
This is something from the old code, so feel free to ignore, but I'm wondering if we have a bug here?

Is `newVisibleH > visibleH || newVisibleW > visibleW` the right heuristic? Imagine two attempts: one that is horrible vertically, but near perfect horizontally, and the other that's near perfect vertically, and horrible horizontally. The only thing that determines which is better* here is the order they're listed — and we're setting things up to prefer the one that comes last in the list, whereas normally when we're giving a list of things to try to a function we list them with the ones we want first.

*assuming they're either both invisible or both visible

Reply:
Yeah I'm not sure on that tbh. I'll log a follow up JIRA to investigate that, as we've already spent a considerable amount of additional time on getting this far.

Reply:
Logged as TINY-7581